packages/query-core/src/__tests__/query.test.tsx

```diff
@@ -965,4 +965,44 @@ describe('query', () => {
     await sleep(60) // let it resolve
     expect(spy).toHaveBeenCalledWith('1 - 2')
   })
+
+  it('should have an error status when queryFn data is not serializable', async () => {
+    const key = queryKey()
+
+    const queryFn = vi.fn()
+
+    queryFn.mockImplementation(async () => {
+      await sleep(1000)
```

Comment:
since we run with real timers:

```suggestion
      await sleep(10)
```